src/components/shared/Validations.tsx

```diff
@@ -202,7 +202,15 @@ export function validationCount(schemaNode: RegularNode) {
 
 export function getValidationsFromSchema(schemaNode: RegularNode) {
   return {
-    ...(schemaNode.enum !== null ? { enum: schemaNode.enum } : null),
+    ...(schemaNode.enum !== null
+      ? { enum: schemaNode.enum }
+      : // in case schemaNode is type: "array", check if its child have defined enum
+      schemaNode.primaryType === 'array' &&
+        schemaNode.children?.length === 1 &&
```

Comment:
tgHmmm, but arrays can also have enums at the top, i.e.

```json
{
  "type": "array",
  "enum": [
    [0, 1],
    [2, 3]
  ],
  "items": {
    "type": "number"
  }
}
```

plus, there's also another "tuples" scenario possible:

```json
{
  "type": "array",
  "items": [
    {
      "type": "number",
      "enum": [0, 1, 2]
    },
    {
      "type": "string",
      "enum": ["a", "b", "c"]
    }
  ]
}
```

Reply:
I knew you will find some cases I overlooked 😄 

If enum is defined for an array, validations will take take that one over child ones, so I think that solved the first thing, right?

I will check what do to with stuff like "touples", right now it wouldn't be supported since I only do this if there is a single item in array

Reply:
> If enum is defined for an array, validations will take take that one over child ones, so I think that solves the first thing, right?

The tricky part is that we should try to display both, because both enums impose certain restrictions.

Example:

```json
{
  "type": "array",
  "enum": [
    ["Maribor", "Zagreb", "Banja Luka", "Skopje"],
    ["Warsaw", "Prague", "Budapest", "Bratislava"]
  ],
  "items": {
    "type": "string",
    "enum": ["Warsaw", "Zagreb", "Prague", "Budapest", "Bratislava", "Maribor"]
  }
}
```

Now, this is valid `["Warsaw", "Prague", "Budapest", "Bratislava"]`, and this is not valid `["Maribor", "Zagreb", "Banja Luka", "Skopje"]`.
The former is valid because all constraints imposed by 2 enums are met, while the latter is invalid because the second enum restricts a certain set of cities.

Does that make sense?

In general, there's no good way to solve this issue. 
I'm just trying to point out the potential problem we might see.

I'm quite convinced that what you implemented here will be good enough for most cases, so we might just take it and merge it.

Reply:
Gotcha, well I would then go with current solution. It is not perfect but neither is what is in prod now. We can then revisit more complex enum scenarios later, if needed.

I'm adding @mnaumanali94 as a reviewer as well, so it is not only my decision 😄

Reply:
Agreed. Lets get this in and as customers find/report complex cases we'll look into it.